File: src/types/prerequisites/single/RatedSumPrerequisite.ts
import {
  Array,
  IncludeIdentifier,
  Integer,
  Object,
  Optional,
  Required,
  TypeAlias,
} from "tsondb/schema/dsl"
import { SkillIdentifier } from "../../_Identifier.js"
import { DisplayOption } from "../DisplayOption.js"

export const RatedSumPrerequisite = TypeAlias(import.meta.url, {
  name: "RatedSumPrerequisite",
  type: () =>
    Object({
      sum: Required({
        comment: "The minimum required sum of the targets’ ratings.",
        type: Integer({ minimum: 0 }),
      }),
      targets: Required({
        comment: "The targets that are included in calculating the sum.",
        type: Array(SkillIdentifier(), { minItems: 2 }),
      }),
      display_option: Optional({
        type: IncludeIdentifier(DisplayOption),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/Types/prerequisites/single/RatedSumPrerequisite.swift
import FileDB

@Embedded
public struct RatedSumPrerequisite {

  /// The minimum required sum of the targets’ ratings.
  @Minimum(0)
  let sum: Int

  /// The targets that are included in calculating the sum.
  @MinItems(2)
  @Relationship(Skill.self)
  let targets: [Skill.ID]

      let display_option: DisplayOption?
  }
